### PR TITLE
Add test case interpolating stack

### LINKED ISSUE #6: Add test case: Create a 3D stack that is the interpolation between two 2D images.

### PR DESCRIPTION
**Pull request by marabuuu**
This PR contains:
* [x] a new test-case for the benchmark
  * [x] I hereby confirm that NO LLM-based technology (such as github copilot) was used while writing this benchmark
* [ ] new dependencies in requirements.txt
  * [ ] The environment.yml file was updated using the command `conda env export > environment.yml`
* [ ] new generator-functions allowing to sample from other LLMs
* [ ] new samples (`sample_....jsonl` files)
* [ ] new benchmarking results (`..._results.jsonl` files)
* [ ] documentation update
* [ ] bug fixes 

Related github issue (if relevant): closes #6 

Short description:
- adds a test case for interpolating 2 2d slices into a stack (inspired by [this discussion](https://stackoverflow.com/questions/36398847/numpy-how-interpolate-between-two-arrays-for-various-timesteps))

How do you think will this influence the benchmark results?
- should be a relatively simple problem

Why do you think it makes sense to merge this PR?
- interpolation is frequently used in BiA

### CODE REVIEW

**Review by haesleinhuepf — COMMENTED** [review]
Great @marabuuu ! Do you think it would large effort to have `n` interpolated slices instead of just one? It would then be a bit more flexible and better reflect practical use-cases.

**Comment by marabuuu** [comment]
Hi @haesleinhuepf ! Thank you for your quick reply. Yes I think it makes sense and I also wanted to add it, but thought it would be cool to have your opinion first: My first idea would be to add another input to the function called num_slices and then in the function: 
If num_slices = 1,
    Do what is already there 
Else:
    Take somehow num_slices into   
    account and interpolate linearly over 
    them.

Does this make sense or should I do it differently? 

Best,
Mara

**Comment by haesleinhuepf** [comment]
If would not write the if-statement. Consider using [np.linspace](https://numpy.org/doc/stable/reference/generated/numpy.linspace.html) (sorry, I should have mentioned earlier).

**Comment by marabuuu** [comment]
Oki thanks, I will do that

**Comment by marabuuu** [comment]
I modified the test case using `np.linspace`, if there is anything else that needs modification I can change it

File: test_cases/interpolate_stack.ipynb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "46340fa1-dcfc-4f44-89c5-b874e1c551b7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def interpolate_stack(start_slice, end_slice, num):\n",
+    "    '''\n",
+    "    Creates a 3D stack that is the interpolation between two 2D slices \n",
```
Comment (haesleinhuepf) [inline test_cases/interpolate_stack.ipynb:12]:
```suggestion
    "    Creates and returns a 3D stack with num planes that is the interpolation between two 2D slices. \n",
```

Comment (haesleinhuepf) [reply test_cases/interpolate_stack.ipynb:12]: How about this @marabuuu ?

**Comment by marabuuu** [comment]
Yes, thank you for the hint! I agreed to this change

File: test_cases/interpolate_stack.ipynb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+   "execution_count": 1,
+   "id": "46340fa1-dcfc-4f44-89c5-b874e1c551b7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def interpolate_stack(start_slice, end_slice, num):\n",
+    "    '''\n",
+    "    Creates and returns a 3D stack with num planes that is the interpolation between two 2D slices. \n",
+    "    '''\n",
+    "    # import statement\n",
```
Comment (haesleinhuepf) [inline test_cases/interpolate_stack.ipynb:14]:
```suggestion
```

File: test_cases/interpolate_stack.ipynb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "1e688885-d20e-477c-ab3c-bec4ebb44584",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def check(candidate):\n",
+    "    #import statement\n",
```
Comment (haesleinhuepf) [inline test_cases/interpolate_stack.ipynb:27]:
```suggestion
```

**Comment by haesleinhuepf** [comment]
Thanks again Mara!

**Pull request merged**